pandas/tests/series/test_arithmetic.py

```diff
@@ -491,14 +491,27 @@ def test_ser_cmp_result_names(self, names, comparison_op):
             result = op(ser, cidx)
             assert result.name == names[2]
 
-    def test_comparisons(self):
+    def test_comparisons(self, using_infer_string):
         s = Series(["a", "b", "c"])
         s2 = Series([False, True, False])
 
         # it works!
         exp = Series([False, False, False])
-        tm.assert_series_equal(s == s2, exp)
-        tm.assert_series_equal(s2 == s, exp)
+        if using_infer_string:
+            import pyarrow as pa
+
+            msg = "has no kernel"
+            # with tm.assert_produces_warning(DeprecationWarning, match="comparison"):
+            with pytest.raises(pa.lib.ArrowNotImplementedError, match=msg):
+                s == s2
```

Comment:
This should probably return all `False` in the future?

Reply:
Yes if we want to patch the arrow behaviour, makes probably sense

Reply:
OK could you add a TODO here as a reminder? Would be good to include it in an issue referencing things that would need fixing before 3.0